### PR TITLE
[new] - Add iframe-has-title rule.

### PR DESCRIPTION
**Pull request by evcohen**
From #29

### CODE REVIEW

File: __tests__/src/rules/frame-has-title-test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+const expectedError = {
+  message: 'Frame elements (frame, iframe) must have a unique title property.',
+  type: 'JSXOpeningElement',
+};
+
+ruleTester.run('html-has-lang', rule, {
+  valid: [
+    { code: '<div />;', parserOptions },
+    { code: '<frame title="Unique title" />', parserOptions },
```
Comment (ljharb) [inline __tests__/src/rules/frame-has-title-test.js]: do people still use `<frame>` elements? do they even work in browsers anymore? i know framesets don't

Comment (evcohen) [inline __tests__/src/rules/frame-has-title-test.js]: eh, doesn't seem so: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/frame

Comment (evcohen) [inline __tests__/src/rules/frame-has-title-test.js]: will change to just iframe

File: src/rules/iframe-has-title.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    JSXOpeningElement: (node) => {
+      const type = elementType(node);
+
+      if (type && type !== 'iframe') {
+        return;
+      }
+
+      const title = getPropValue(getProp(node.attributes, 'title'));
+
+      if (title) {
```
Comment (ljharb) [inline src/rules/iframe-has-title.js]: does this mean that the following will fail the rule?
```js
title=""
title={false}
title={''}
title={``}
```
If so, it'd be nice to include this edge case  in the docs and tests.

Comment (evcohen) [inline src/rules/iframe-has-title.js]: yes these will fail - `title={true}` will pass, which is unfortunate. we can use getLiteralPropValue here and bypass the boolean edge case, but then we'd be failing on `title={foo}` where `foo` is some variable. we can also explicitly check `title && title !== true` if that seems more appropriate?

Comment (ljharb) [inline src/rules/iframe-has-title.js]: I think that seems reasonable; I mainly want it documented and to have test cases :-)
